scripts/release/asset_publish.py

```diff
@@ -304,6 +375,29 @@ def get_asset_details(
     return json.loads(result.stdout)
 
 
+def get_parsed_details_from_asset_uri(asset_type: str, asset_uri: str) -> Tuple[str, str, str, str]:
+    """Validate asset URI and return parsed details. Exception is raised for an invalid URI.
+
+    :param asset_type: Valid values are component, environment and model
+    :type asset_type: str
+    :param asset_uri: A workspace or registry asset URI to parse
+    :type asset_uri: str
+    :return:
+        A tuple with asset `name`, `version`, `label`, and `registry_name` in order.
+        `label` and `registry_name` will be None for workspace URI.
+    :rtype: Tuple
+    """
+    REGISTRY_ASSET_PATTERN = re.compile(REGISTRY_ASSET_TEMPLATE.substitute({ASSET_TYPE: asset_type}))
```

Comment:
Simpler to use kwargs:
```suggestion
    REGISTRY_ASSET_PATTERN = re.compile(REGISTRY_ASSET_TEMPLATE.substitute(asset_type=asset_type))
```

Reply:
Done